ft_split.c

```diff
@@ -0,0 +1,55 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: bchristo <bchristo@student.42wolfsburg.    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2021/05/26 21:10:01 by bchristo          #+#    #+#             */
+/*   Updated: 2021/05/27 03:37:22 by bchristo         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+static int	ft_isspace(int c)
+{
+	if (c == '\t' || c == '\n' || c == '\v' || c == '\f' || c == '\r' || c == ' ')
+		return (1);
+	return (0);
+}
+
+static void	init(char **r, size_t *i)
+{
+	r[0] = NULL;
+	*i = 0;
+}
+
+char	**ft_split(char const *s, char c)
+{
+	char		**r;
+	char		*end;
+	size_t		i;
+
+	r = (char **)malloc((ft_strlen(s) + 1) * sizeof(char *));
```

Comment:
He is right. For this one, you actually didn't allocate memory for each trimmed string inside **r. You need to allocate memory for **r, and loop through it to allocate memory for each string stored inside. [https://stackoverflow.com/questions/19240540/dynamically-allocating-array-explain/19240932#19240932](url)

Reply:
isnt this what line 24 does?
```c
	ft_strlcpy(sub, s + start, len);
```